### PR TITLE
[User & Group Manager] migrate legacy user from ns v1 to ns v2.

### PR DESCRIPTION
**Pull request by ydye**

### CODE REVIEW

File: src/rest-server/deploy/legacy_user_migrate.py [hunk 1]
```diff
@@ -53,6 +57,16 @@ def etcd_data_parse(self):
             sys.exit(1)
         return user_list
 
+    def namespace_v1_data_prepare(self):
+        nsv1_result = http_get(self.k8s_conn, '/api/v1/namespaces/pai-user/secrets/', {'Accept': 'application/json'})
+        if nsv1_result['code'] == 200:
+            return json.loads(nsv1_result['data']['items'])
+        elif etcd_result['code'] == 404:
```
Comment (wangdian) [inline src/rest-server/deploy/legacy_user_migrate.py]: Should be nsv1_result['code']?

File: src/rest-server/deploy/legacy_user_migrate.py [hunk 2]
```diff
@@ -19,14 +19,18 @@ def __init__(self, user_name):
...
         self.etcd_uri = etcd_uri
         self.k8s_uri = k8s_uri
+        self.admin_group = admin_groupname
         self.etcd_conn = http.client.HTTPConnection(self.etcd_uri)
         self.k8s_conn = http.client.HTTPConnection(self.k8s_uri)
         self.flag_path = '/v2/keys/transferFlag'
         self.etcd_prefix = '/users/'
         self.secret_ns = "pai-user"
+        self.secret_ns_user_V2 = "pai-user-v2"
+        self.secret_ns_group_v2 = "pai-group"
```
Comment (wangdian) [inline src/rest-server/deploy/legacy_user_migrate.py]: Please uniform word cases. V2 should be lower case.
secret_ns_group_v2 should be secret_ns_group

File: src/rest-server/deploy/legacy_user_migrate.py [hunk 3]
```diff
@@ -103,6 +194,23 @@ def create_secret_user(self, payload):
...
+                logger.error("Create user in k8s secret failed")
+                sys.exit(1)
+
+    def create_secret_group_v2(self, payload):
+        check_res = http_get(self.k8s_conn, '/api/v1/namespaces/{0}/secrets/{1}'.format(self.secret_ns_group_v2,payload['metadata']['name']))
+        if check_res['code'] == 404:
+            post_res = http_post(self.k8s_conn, '/api/v1/namespaces/{0}/secrets/'.format(self.secret_ns),
+                             json.dumps(payload))
+            if post_res['code'] != 201:
+                logger.error("Create user in k8s secret failed")
```
Comment (wangdian) [inline src/rest-server/deploy/legacy_user_migrate.py]: Should be "Create group in k8s ..."

File: src/rest-server/deploy/legacy_user_migrate.py [hunk 4]
```diff
@@ -53,6 +57,16 @@ def etcd_data_parse(self):
             sys.exit(1)
         return user_list
 
+    def namespace_v1_data_prepare(self):
+        nsv1_result = http_get(self.k8s_conn, '/api/v1/namespaces/pai-user/secrets/', {'Accept': 'application/json'})
+        if nsv1_result['code'] == 200:
+            return json.loads(nsv1_result['data']['items'])
+        elif etcd_result['code'] == 404:
+            logger.info("No ledacy user data found in k8s namespace pai-user")
```
Comment (wangdian) [inline src/rest-server/deploy/legacy_user_migrate.py]: Missing return value or exit code.

File: src/rest-server/deploy/legacy_user_migrate.py [hunk 5]
```diff
@@ -79,6 +93,52 @@ def secret_data_prepare(self, user_info):
...
+            'extension': str(base64.b64encode(json.dumps(extension)), 'utf-8'),
+        }
+
+        post_data_dict = {}
+        post_data_dict['metadata'] = meta_dict
+        post_data_dict['data'] = user_dict
+
+        return post_data_dict
+
+    def secret_data_prepare_v2_group(self, groupname):
```
Comment (wangdian) [inline src/rest-server/deploy/legacy_user_migrate.py]: This function is create group for VC, can not see from api. Should be secret_data_prepare_vc_group?

File: src/rest-server/deploy/legacy_user_migrate.py [hunk 6]
```diff
@@ -160,27 +272,39 @@ def main():
...
+      logger.info("Etcd data has already been transferred to k8s secret")
 
-    logger.info('Legacy user data transfer from etcd to kubernetes secret successfully')
+    res = transferCli.prepare_secret_base_path_v2()
+    if res[0] == 1 and res[1] == 1:
+      ns_pai_user_list = transferCli.namespace_v1_data_prepare()
+      for user in ns_pai_user_list:
+          secret_post_data = transferCli.secret_data_prepare_v2(user)
+          transferCli.create_secret_user_v2(secret_post_data)
+      vc_set = transferCli.vc_set
```
Comment (wangdian) [inline src/rest-server/deploy/legacy_user_migrate.py]: Not sure if this function may miss any vc

**Comment by ydye** [comment]
@mzmssg How to get all vc list in a cluster

File: src/rest-server/deploy/legacy_user_migrate.py (hunk 2 above)
Comment (ydye) [reply src/rest-server/deploy/legacy_user_migrate.py]: Done

File: src/rest-server/deploy/legacy_user_migrate.py (hunk 4 above)
Comment (ydye) [reply src/rest-server/deploy/legacy_user_migrate.py]: Done

**Pull request merged**